File: Cpp/1337_The_K_Weakest_Rows_in_a_Matrix.cpp
class Solution {
public:
    vector<int> kWeakestRows(vector<vector<int>>& mat, int k) {
        vector<int> result(k);
        priority_queue<pair<int,int>> pq;
        
        for(int i=0; i<mat.size(); ++i){
            int count = 0;
            for(int j=0; j<mat[0].size(); ++j){
                if(mat[i][j] == 0) break;
                ++count;
            }
            pq.push({count, i});
            if(pq.size() > k) pq.pop();
        }
        
        for(int i=k-1; i>=0; --i){
            result[i] = pq.top().second;
            pq.pop();
        }
        
        return result;
    }
};

File: Kotlin/1337_The_K_Weakest_Rows_in_a_Matrix.kt
class Solution {
    fun kWeakestRows(mat: Array<IntArray>, k: Int): IntArray {
        val result = IntArray(k)
        val pq = PriorityQueue<Pair<Int, Int>>() { a, b ->
            if(a.first == b.first) b.second - a.second
            else b.first - a.first
        }
        
        for(i in 0 until mat.size) {
            pq.offer(mat[i].sum() to i)
            if(pq.size > k) pq.poll()
        }
        
        for(i in 0 until k){
            result[k - 1 - i] = pq.poll().second
        }
        
        return result
    }
}